### PR TITLE
Feature/corresponding mat transpose inline function

### PR DESCRIPTION
**Pull request by csmichael-as**
In place functions for matrix multiplication and matrix transposition.

### CODE REVIEW

File: inc/xmath.h [hunk 1]
```diff
@@ -435,9 +434,20 @@ static inline void vec_hadamard(size_t n, TYPE dst[n], TYPE left[n], TYPE right[
...
 
-static inline TYPE vec_mag(size_t n, TYPE left[n])
-{ VEC_MAG(TYPE, n, left) }
+static inline void mat_transpose(size_t r, size_t c, TYPE in[r][c], TYPE out[c][r])
+{ MAT_TRANSPOSE(TYPE, r, c, in, out) }
+
+static inline void mat_square_multiplication(size_t n, TYPE m[n][m_C], TYPE n[n][n], TYPE resultant[n][n]) 
+{ MAT_MUL(TYPE, n, m_C, n, n, resultant, m, n) }
 
+static inline void mat_multiplication(size_t m_R,
```
Comment (mrpossoms) [inline inc/xmath.h:443]: `mat_multiplication` is really verbose. At that point you might as well make it `matrix_multiplication`. Imagine typing that over and over again when your implementing some equation.

File: inc/xmath.h [hunk 2]
```diff
@@ -435,9 +434,20 @@ static inline void vec_hadamard(size_t n, TYPE dst[n], TYPE left[n], TYPE right[
...
+                                        size_t m_C,
```
Comment (mrpossoms) [inline inc/xmath.h:444]: `resultant` doesn't follow the same naming convention of the operands m and n. It's just kinda strange when some parameters are extremely terse, and others are verbose.

File: inc/xmath.h (hunk 1 above)
Comment (mrpossoms) [inline inc/xmath.h:443]: I wonder if moving some of the args around might make this function easier to use. Something like 
m_r, m_c, M, n_r, n_c, N.

File: inc/xmath.h [hunk 3]
```diff
@@ -435,9 +434,20 @@ static inline void vec_hadamard(size_t n, TYPE dst[n], TYPE left[n], TYPE right[
 static inline TYPE vec_dot(size_t n, TYPE left[n], TYPE right[n])
 { VEC_DOT(TYPE, n, left, right) }
 
-static inline TYPE vec_mag(size_t n, TYPE left[n])
-{ VEC_MAG(TYPE, n, left) }
+static inline void mat_transpose(size_t r, size_t c, TYPE in[r][c], TYPE out[c][r])
+{ MAT_TRANSPOSE(TYPE, r, c, in, out) }
+
+static inline void mat_square_multiplication(size_t n, TYPE m[n][m_C], TYPE n[n][n], TYPE resultant[n][n])
```
Comment (mrpossoms) [inline inc/xmath.h:440]: I don't think we need this. It helps save 3 params, but I'm not sure it's worth it. If we do keep it, a more terse function name would be preferable.

File: tests/Makefile [hunk 4]
```diff
@@ -14,7 +14,7 @@ bin:
 bin/%: %.cpp bin
 	$(CXX) $(CFLAGS) $(INC) $(LIBS) $< $(SRC_OBJS) -o $@ $(LINK)
```
Comment (mrpossoms) [inline tests/Makefile:15]: If you're adding CPPFLAGS, shouldn't  the CPP rule used them?

Comment (csmichael-as) [reply tests/Makefile:15]: that was made in error

File: inc/xmath.h (hunk 3 above)
Comment (csmichael-as) [reply inc/xmath.h:440]: :+1:

File: inc/xmath.h (hunk 1 above)
Comment (csmichael-as) [reply inc/xmath.h:443]: :+1:

File: inc/xmath.h (hunk 2 above)
Comment (csmichael-as) [reply inc/xmath.h:444]: :brain:

File: inc/xmath.h (hunk 1 above)
Comment (csmichael-as) [reply inc/xmath.h:443]: :+1:

**Pull request merged**